src/script/backport-create-issue

```diff
@@ -305,20 +305,34 @@ def mark_as_processed(r, issue):
     This script will add a custom Tag to indicate whether the tracker was
     already processed for backport tracker creation.
     """
-    custom_fields = list(issue['custom_fields'].values())
-    for i, field in enumerate(custom_fields):
-        if field['name'] == 'Tags':
-            if tag_backport_processed not in field['value']:
-                if field['value']:
-                    custom_fields[i]['value'] += (tag_separator +
-                                                  tag_backport_processed)
-                else:
-                    custom_fields[i]['value'] = tag_backport_processed
-                logging.info("%s adding tag '%s'", url(issue),
-                             tag_backport_processed)
-                r.issue.update(issue.id, custom_fields=custom_fields)
-                return
 
+    logging.debug("custom_fields: %s", list(issue['custom_fields']))
+
+    tags_cf = next(filter(lambda x: x['id'] == CF_TAGS, issue['custom_fields']), None)
+
+    if tags_cf is None:
+        tags = ''
+    else:
+        tags = tags_cf.value
+        if tags is None:
+            tags = ''
+        else:
+            tags.strip()
+
+    if TAG_BACKPORT_PROCESSED not in tags:
+        if tags:
+            tags += f"{TAG_SEPARATOR}{TAG_BACKPORT_PROCESSED}"
+        else:
+            tags = TAG_BACKPORT_PROCESSED
+
+        logging.info("%s adding tag '%s', now '%s'", url(issue), TAG_BACKPORT_PROCESSED, tags)
+        tags_cf = {
+          'id': CF_TAGS,
+          'value': tags,
+        }
+        r.issue.update(issue.id, custom_fields=[tags_cf])
```

Comment:
Previously, the entire `custom_fields` list was passed here, not just the field that we are interested in updating.  Does Redmine preserve other custom fields on the ticket in this case?

Reply:
Yes.